docs/design/datacontracts/Thread.md

```diff
@@ -69,14 +69,19 @@ The contract additionally depends on these data descriptors
 
 | Data Descriptor Name | Field | Meaning |
 | --- | --- | --- |
+| `Exception` | `WatsonBuckets` | Pointer to exception Watson buckets |
 | `ExceptionInfo` | `PreviousNestedInfo` | Pointer to previous nested exception info |
+| `ExceptionInfo` | `ThrownObjectHandle` | Pointer to exception object handle |
+| `ExceptionInfo` | `ExceptionWatsonBucketTrackerBuckets` | Pointer to Watson unhandled buckets |
 | `GCAllocContext` | `Pointer` | GC allocation pointer |
 | `GCAllocContext` | `Limit` | Allocation limit pointer |
 | `IdDispenser` | `HighestId` | Highest possible small thread ID |
 | `IdDispenser` | `IdToThread` | Array mapping small thread IDs to thread pointers |
 | `InflightTLSData` | `Next` | Pointer to next in-flight TLS data entry |
 | `InflightTLSData` | `TlsIndex` | TLS index for the in-flight static field |
 | `InflightTLSData` | `TLSData` | Object handle to the TLS data for the static field |
+| `MethodTable` | `BaseSize` | Base size of instance of this class |
+| `Object` | `m_pMethTab` | Pointer to the object's method table |
```

Comment:
We won't need to directly refer to the object and MethodTable data members if we rely on the Object contract instead.